### DISCUSSION TITLE
How to obtain the initial texture file of VPSC by processing the XRD measurement data

### DISCUSSION THREAD

**Discussion by ddyyasfh** [mtex-toolbox/mtex#1767] [Ask Anything] [answered]
Hi everyone

How to obtain the initial texture file of VPSC by processing the XRD measurement data.

Thank you,
Dy.

**Comment by lfgmorales** [comment, marked as answer]
Not sure if I understood this question correctly, but you may potentially calculate an odf out of your pole figure data and use the command discreteSample to create your initial texture file?

Have a look here

<https://mtex-toolbox.github.io/discreteSample.html>
discreteSample | MTEX<https://mtex-toolbox.github.io/discreteSample.html>
mtex-toolbox.github.io<https://mtex-toolbox.github.io/discreteSample.html>
        [favicon.png] <https://mtex-toolbox.github.io/discreteSample.html>



Cheers

L

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 14 Mar 2023, at 17:14, Crazetopher Dvaid Blazing IV ***@***.***> wrote:

﻿

Hi everyone

How to obtain the initial texture file of VPSC by processing the XRD measurement data.

Thank you,
Dy.

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1767>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5INZIMRLBX72K6R7KIHDW4CKO7ANCNFSM6AAAAAAV2VEA7M>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1767", "url": "https://github.com/mtex-toolbox/mtex/discussions/1767", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by ddyyasfh** [reply]
Thank you for the suggestion, I calculated the odf number by

>  odf = calcODF(pf,'silent')

 after importing the xrd data, but I prompted 
_Undefined function or variable 'points', 'f'_  when using the following command.               
`  x = discreteSample(f,points)`

**Comment by lfgmorales** [comment]
If you type help discreteSample, you will see that f=odf


Cheers

L

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 15 Mar 2023, at 07:24, Crazetopher Dvaid Blazing IV ***@***.***> wrote:

﻿

Thank you for the suggestion, I calculated the odf number by 'odf = calcODF(pf,'silent')' after importing the xrd data, but I prompted
Undefined function or variable 'points', 'f' when using the following command.
x = discreteSample(f,points)

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1767#discussioncomment-5317634>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN7IULDIG5MVWOOUNXDW4FN7VANCNFSM6AAAAAAV2VEA7M>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1767#discussioncomment-5317634", "url": "https://github.com/mtex-toolbox/mtex/discussions/1767#discussioncomment-5317634", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by ddyyasfh** [reply]
But when I enter f, it says undefined, but change to odf will do

**Reply by ddyyasfh** [reply]
Points refers to the number of random samples, how to confirm this number, I got 4942 odf data, but I entered points can exceed this number, and the Inv. column data in the Euler angle data is all 0.
![1](https://user-images.githubusercontent.com/104597723/225242674-4d2befe1-ea16-427c-977a-88903fbc1d5d.png)

**Comment by lfgmorales** [comment]
I don’t understand what you want to do, but if you need 1000 points, than write x=discreteSample(odf,1000), for vpsc you don’t need the last column, you only need the orientations



Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Reply by ddyyasfh** [reply]
Doesn't the last column of VPSC need weights? Below is my ODF data, of which there are 4924.
![2](https://user-images.githubusercontent.com/104597723/225248380-084a6549-f26b-4709-ad78-33e029c277a9.png)

**Comment by lfgmorales** [comment]
<https://public.lanl.gov/lebenso/VPSC7c_manual.pdf>
public.lanl.gov<https://public.lanl.gov/lebenso/VPSC7c_manual.pdf>
        [X] <https://public.lanl.gov/lebenso/VPSC7c_manual.pdf>



Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 15 Mar 2023, at 09:19, Crazetopher Dvaid Blazing IV ***@***.***> wrote:

﻿

Doesn't the last column of VPSC need weights? Below is my ODF data, of which there are 4924.
[2]<https://user-images.githubusercontent.com/104597723/225248380-084a6549-f26b-4709-ad78-33e029c277a9.png>

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1767#discussioncomment-5318517>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN7GLKOCFEBEXKS2IOTW4F3RZANCNFSM6AAAAAAV2VEA7M>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1767#discussioncomment-5318517", "url": "https://github.com/mtex-toolbox/mtex/discussions/1767#discussioncomment-5318517", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by ddyyasfh** [reply]
Thank you very much for your help, it has been very rewarding,Best wishes to you.
